src/main/kotlin/org/ionproject/integration/ui/controller/JobController.kt

```diff
@@ -29,30 +32,47 @@ class JobController(
         @RequestBody body: CreateJobDto,
         servletRequest: HttpServletRequest,
         response: HttpServletResponse
-    ): String {
+    ): PostResponse {
         val request = inputProcessor.getJobRequest(body)
         val requestResult = jobEngine.runJob(request)
 
         return when (requestResult.result) {
             JobEngine.JobExecutionResult.CREATED -> {
-                response.status = HttpServletResponse.SC_CREATED
-                response.addHeader("Location", servletRequest.getLocationForJobRequest(requestResult))
-                "Created ${request.javaClass.simpleName} job with ID ${requestResult.jobId}"
+                PostResponse(
+                    location = servletRequest.getLocationForJobRequest(requestResult),
+                    status = HttpStatus.CREATED,
+                    response = response
+                )
             }
             else -> {
                 logger.error("Job creation failed: $body")
-                response.status = HttpServletResponse.SC_BAD_REQUEST
-                "FAILED: ${requestResult.result}"
+                PostResponse(
+                    status = HttpStatus.BAD_REQUEST,
+                    response = response
+                )
             }
         }
     }
 
     @GetMapping
-    fun getJobs(): List<JobEngine.IntegrationJob> = jobEngine.getRunningJobs()
+    fun getJobs(servletRequest: HttpServletRequest): List<JobDetailDto> =
+        jobEngine.getRunningJobs()
+            .map { job ->
+                val url = servletRequest.getLocationForJobRequest(job.status)
+                JobDetailDto.of(job, url, JobDetailDto.DetailType.METADATA_ONLY)
+            }
 
     @GetMapping("/{id}")
-    fun getJobDetails(@PathVariable id: Long): JobEngine.IntegrationJob = jobEngine.getJob(id)
+    fun getJobDetails(
+        @PathVariable id: Long,
+        servletRequest: HttpServletRequest
+    ): JobDetailDto {
+        val job = jobEngine.getJob(id)
+        val url = servletRequest.getLocationForJobRequest(job.status)
+
+        return JobDetailDto.of(job, url, JobDetailDto.DetailType.FULL)
+    }
 
     private fun HttpServletRequest.getLocationForJobRequest(jobStatus: JobEngine.JobStatus): String =
-        "$localName:${localPort}$JOBS_URI/${jobStatus.jobId}"
+        "$scheme://$serverName:${localPort}$contextPath$JOBS_URI/${jobStatus.jobId}"
```

Comment:
if ${localPort} = 80 then can be ignored.

Reply:
done